### ISSUE TITLE
Batchnorm's running_var is wrong(I test it use momentum = 1)

### ISSUE THREAD

**Issue by zhujianing** [pytorch/pytorch#7028]
## Issue description

I test BN1d use momentum = 1, but i found  running_var is not equal to `numpy.std(input.data.numpy()`'s output , but runing_mean is right , it is equal to `numpy.mean(input.data.numpy()` 
## there is my testing code
```
import torch
from torch import nn
import numpy
from torch.autograd import Variable

bn = nn.BatchNorm1d(4,momentum=1)
input = Variable(torch.Tensor([ [2,15,2,1] ,[2,17,2,1] ,[-4,-2,-2,-1] ]))
bn.weight.data = torch.ones(4)
bn.bias.data = torch.zeros(4)
bn.training = True
print(bn.train)
output = bn(input)
print(bn._buffers)
print(numpy.mean(input.data.numpy(), axis=0))
print(numpy.std(input.data.numpy(), axis=0))
```
## there are results
```
<bound method BatchNorm1d.train of BatchNorm1d(4, eps=1e-05, momentum=1, affine=True, track_running_stats=True)>
OrderedDict([('running_mean', tensor([  0.0000,  10.0000,   0.6667,   0.3333])), ('running_var', tensor([  12.0000,  109.0000,5.3333,    1.3333]))])
[ 0.         10.          0.6666667   0.33333334]
[2.828427   8.524474   1.8856181  0.94280905]
```

**Comment by crcrpar** [comment]
Anyway, `numpy.std` and `numpy.var` divide sum of square by `n` (see [numpy document](https://docs.scipy.org/doc/numpy/reference/generated/numpy.std.html)), 
otoh, pytorch divide sum of square by `n-1` ([source](https://github.com/pytorch/pytorch/blob/master/aten/src/THNN/generic/BatchNormalization.c#L54-L58)).
So, there should be some differences.

**Comment by SsnL** [comment]
PyTorch BN is correct. It's the [unbiased estimator of variance](https://en.wikipedia.org/wiki/Bias_of_an_estimator#Sample_variance), which is the desired one for BN. If you calculate the unbiased np std, it will be the same. See `ddof` option of `np.std`.
